frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/utils.ts

```diff
@@ -260,7 +264,7 @@ export const connectFinallyTasksToNodes = (
   if (!pipeline.spec?.finally) {
     return nodes;
   }
-  const finallyTasks = pipeline.spec.finally;
+  const finallyTasks = getFinallyTasks(pipeline, pipelineRun);
```

Comment:
based on my other comment
```suggestion
const finallyTasks = pipelineRun ? getFinallyTasksWithStatus(pipeline, pipelineRun) : pipeline.spec.finally;
```